version: '3.8'

services:
  postgres:
    image: postgres:alpine
    environment:
      POSTGRES_DB: pilip
      POSTGRES_PASSWORD: password
      POSTGRES_USER: username
    ports:
      - 5435:5432
    restart: on-failure:3
    networks:
      - backend-network
  redis:
    image: redis:latest
    command: redis-server
    volumes:
      - redis:/var/lib/redis
      - redis-config:/usr/local/etc/redis/redis.conf
    ports:
      - 6379:6379
    networks:
      - backend-network
  backend:
    image: python:3.10
    build: 
      context: path-to-backend-folder
      dockerfile: Dockerfile
    ports:
      - 5656:5656
    environment:
      - DEBUG=true
      - DB_URL=postgres://root:root@postgres:5432/pilip
      - REDIS_URL=redis://redis:6379/
      - AUTH=root:root,groot:groot
    restart: on-failure:3
    depends_on:
      - redis
      - postgres
    networks:
      - backend-network
volumes:
  redis:
  redis-config:
networks:
  backend-network:
    driver: bridge